kartothek/serialization/_generic.py

```diff
@@ -266,28 +280,28 @@ def _ensure_type_stability(array_like, value, strict_date_types, require_ordered
 
     value_dtype = pd.Series(value).dtype
     array_like, array_value_type = _handle_categorical_data(array_like, require_ordered)
-    array_like, array_value_type = _handle_null_arrays(array_like, value_dtype)
+    array_value_type = _handle_bool_arrays(array_like, array_value_type, value_dtype)
+    array_like, array_value_type = _handle_null_arrays(
+        array_like, array_value_type, value_dtype
+    )
 
-    type_comp = (value_dtype.kind, array_value_type.kind)
+    type_comp = {value_dtype.kind, array_value_type.kind}
 
     compatible_types = [
         # UINT and INT
-        ("u", "i"),
-        ("i", "u"),
+        {"u", "i"},
         # various string kinds
-        ("O", "S"),
-        ("O", "U"),
-        # bool w/ Nones
-        ("b", "O"),
+        {"S", "O"},
+        {"U", "O"},
```

Comment:
Would this allow the comparison of an array of non-string objects with a string?